components/function-controller/pkg/controller/function/function_controller.go

```diff
@@ -106,385 +98,436 @@ var (
+		log.Error(err, "Error creating RuntimeInfo", "namespace", fnConfig.Namespace, "name", fnConfig.Name)
 		return reconcile.Result{}, err
 	}
 
-	// Create Function's ConfigMap
-	foundCm := &corev1.ConfigMap{}
-	deployCm := &corev1.ConfigMap{}
-	_, err = r.createFunctionConfigMap(foundCm, deployCm, fn)
+	// Synchronize Function ConfigMap
+	fnCm, err := r.syncFunctionConfigMap(fn)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return reconcile.Result{}, nil
+		if err := r.updateFunctionStatus(fn, serverlessv1alpha1.FunctionConditionError); err != nil {
+			log.Error(err, "Error setting Function status", "namespace", fn.Namespace, "name", fn.Name)
 		}
-		r.updateFunctionStatus(fn, serverlessv1alpha1.FunctionConditionError)
 
-		log.Error(err, "function configmap can't be created. The function could have been deleted.", "namespace", deployCm.Namespace, "name", deployCm.Name)
+		log.Error(err, "Error during sync of the Function ConfigMap", "namespace", fn.Namespace, "name", fn.Name)
 		return reconcile.Result{}, err
 	}
 
-	// Update Function's ConfigMap
-	if err := r.updateFunctionConfigMap(foundCm, deployCm); err != nil {
-		// status of the functon must change to error.
-		r.updateFunctionStatus(fn, serverlessv1alpha1.FunctionConditionError)
+	// Synchronize Function build Task
+	if err := r.syncFunctionBuildTask(fn, rnInfo); err != nil {
+		if err := r.updateFunctionStatus(fn, serverlessv1alpha1.FunctionConditionError); err != nil {
+			log.Error(err, "Error setting Function status", "namespace", fn.Namespace, "name", fn.Name)
+		}
 
-		log.Error(err, "Error while trying to update Function's ConfigMap:", "namespace", deployCm.Namespace, "name", deployCm.Name)
+		log.Error(err, "Error during sync of the Function build Task", "namespace", fn.Namespace, "name", buildTaskName)
 		return reconcile.Result{}, err
 	}
 
-	// Create function's image name
-	hash := sha256.New()
-	hash.Write([]byte(foundCm.Data["handler.js"] + foundCm.Data["package.json"]))
-	functionSha := fmt.Sprintf("%x", hash.Sum(nil))
-	imageName := fmt.Sprintf("%s/%s-%s:%s", rnInfo.RegistryInfo, fn.Namespace, fn.Name, functionSha)
-	log.Info("function image", "namespace:", fn.Namespace, "name:", fn.Name, "imageName:", imageName)
-
-	if err := r.getFunctionBuildTemplate(fn, rnInfo); err != nil {
-		// status of the functon must change to error.
-		r.updateFunctionStatus(fn, serverlessv1alpha1.FunctionConditionError)
-
+	// Generate build and image names
+	fnSha, err := generateFunctionHash(fnCm)
+	if err != nil {
+		log.Error(err, "Error generating Function hash", "namespace", fn.Namespace, "name", fn.Name)
 		return reconcile.Result{}, err
 	}
 
-	// Unique Build name base on function sha
-	shortSha := ""
-	if len(functionSha) > 10 {
-		shortSha = functionSha[0:10]
-	} else if len(functionSha) > 0 && len(functionSha) < 10 {
-		shortSha = functionSha
+	shortSha := fnSha
+	if len(fnSha) > 10 {
```

Comment:
not that it matters, but this check will never fail. sha256 always require 64characters in hex representation

Reply:
Nice catch

Reply:
See https://github.com/kyma-project/kyma/pull/5664/commits/7abd9284ecc4c9a2e81ec8f34116b15c3a863e80